### PR TITLE
For #17802: Match the design for three-dot menu navigation

### PR DESCRIPTION
**Pull request by eliserichards**
For #17802 
Wait for https://github.com/mozilla-mobile/fenix/pull/17838 to be merged (✅ merged)

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Review by gabrielluong — CHANGES_REQUESTED** [review]
It was a bit hard for me to tell what the changes were about based on the issue name. Could we also reword it to something like: Move the menu navigation toolbar based on the toolbar orientation

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt [hunk 1]
```diff
@@ -471,29 +476,50 @@ class DefaultToolbarMenu(
...
+                if (syncedTabsInTabsTray) null else syncedTabsItem,
+                BrowserMenuDivider(),
+                findInPageItem,
+                desktopSiteItem,
+                BrowserMenuDivider(),
+                addToHomeScreenItem.apply { visible = ::canAddToHomescreen },
+                addToTopSitesItem,
+                saveToCollectionItem,
+                BrowserMenuDivider(),
+                settingsItem
```
[the remark this answers is not in the document]
Comment (eliserichards) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:499]: The thought here was that the other one will just be deleted when we remove the feature flag, so there would be the minimal amount of effort needed later. What do you think? I'm happy to do it either way

**Comment by eliserichards** [comment]
@gabrielluong the acceptance criteria in the issue is pretty extensive, you can see what changes were made there

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (gabrielluong) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:499]: The list I was referring to was the one in the `else` below. I know we will want to delete the oldMenuItem collection when the feature flag is removed, but I do think it makes it a bit more clear if we inline the checks to this list.

**Comment by gabrielluong** [comment]
> @gabrielluong the acceptance criteria in the issue is pretty extensive, you can see what changes were made there

I can only note 2 particular things that the PR does from looking at the code:
- Moving the menu navigation toolbar based on the toolbar orientation
- Removing the bookmark from the menu navigation toolbar

Since the latter is rather minor, I think the suggested naming of the issue of just addressing the first issue still makes the primary headline. We can add the latter point into the commit description as well or have it as a separate commit. I think I am still looking for the original requested changes?

**Review by gabrielluong — COMMENTED** [review]
I think I am still waiting for the changes on the original feedback?

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (eliserichards) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:499]: Ahh yes, I see what you mean. I was confusing it with another thing.

**Comment by eliserichards** [comment]
I think naming it as matching the design will point them to the issue so they can see the fine details. This naming encompasses all of the changes, instead of just mentioning one change

**Review by gabrielluong — APPROVED** [review]
Looks good, thank you for the changes!

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt [hunk 2]
```diff
@@ -63,9 +63,10 @@ class DefaultToolbarMenu(
     val isPinningSupported: Boolean
 ) : ToolbarMenu {
 
-    private var currentUrlIsBookmarked = false
+    private var isCurrentUrlBookmarked = false
     private var isBookmarkedJob: Job? = null
-
+    private val isTopToolbarSelected = shouldReverseItems
+
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:69]:
```suggestion

```
